subprojects/core/src/main/java/org/gradle/execution/plan/SingleFileTreeElementMatcher.java

```diff
@@ -131,7 +131,13 @@ public int getMode() {
 
         @Override
         public FilePermissions getPermissions() {
-            int unixNumeric = stat.getUnixMode(file);
+            File target;
+            try {
+                target = file.getCanonicalFile();
+            } catch (IOException e) {
+                target = file;
+            }
```

Comment:
❓ Is it reasonable to handle the exception like this? Are we maybe losing interesting context here?

```text
     * @throws  IOException
     *          If an I/O error occurs, which is possible because the
     *          construction of the canonical pathname may require
     *          filesystem queries
```

So essentially the `IOException` is likely to come because something along the canonical path either doesn't exist or isn't accessible. Why is falling back to using the non-canonical path a good alternative in this case? Aren't we falling back to the incorrect behavior then?

Reply:
If there is a severe problem, then it should manifest during the actual copying too, right? And these cases *should* be covered but all sorts of warnings/errors already.

If it's not a severe problem we're falling back to having a file copied with the wrong permissions. Sure, it may lead to problems in some cases, but isn't it a garbage-in - garbage out situation? 

Do you have a particular scenario in mind when this kind of fallback is not reasonable?

Reply:
I'm just uncomfortable with accepting an unknown situation here. I see no harm in throwing the exeception here instead. Is there any reason why we shouldn't?

Reply:
Note: that this thread is obsolete with the latest changes.